python_data_analytics/psql/data_explore.sql

```
@@ -0,0 +1,45 @@
+-- Show table schema 
+\d+ retail;
+
+-- Show first 10 rows
+SELECT * 
+FROM retail 
+limit 10;
+
+-- Check # of records
+select COUNT(*) as num_records
+from retail;
+
+-- number of clients (e.g. unique client ID)
+select count(distinct customer_id) as num_clients
+from retail;
+
+--invoice date range (e.g. max/min dates)
+select max(invoice_date) as max, 
+	   min(invoice_date) as min
+from retail;
+
+--number of SKU/merchants (e.g. unique stock code)
+select count(distinct stock_code) as count
+from retail;
+
+--Calculate average invoice amount excluding invoices with a negative amount 
+--(e.g. canceled orders have negative amount)
+with cte as(SELECT invoice_no, quantity * unit_price as price
+FROM retail),
+cte2 as(select invoice_no,sum(price) as sums
+from cte
+group by 1)
```

avoid using 'group by 1' in production code: it's not always portable to other dbs, and it can fail silently if the select statement changes.